### PR TITLE
Unknown command

### PR DESCRIPTION
**Pull request by dveeden**
```
WARN[0000] unknown command                               arguments=" /tmp/a /tmp/b" command=copy_file line=1
```

### CODE REVIEW

**Comment by dveeden** [comment]
cc @morgo @mjonss

**Comment by morgo** [comment]
This is likely to break the existing tests, since many have macros in them that are not supported.

**Comment by mjonss** [comment]
Did you mean to support `--remove_file` or just add logging of unsupported commands?
If only log (normally to stdout) non-supported commands, then it LGTM
Otherwise you reverted the first commit which also added support for `--remove_file`

**Comment by dveeden** [comment]
> Did you mean to support `--remove_file` or just add logging of unsupported commands? If only log (normally to stdout) non-supported commands, then it LGTM Otherwise you reverted the first commit which also added support for `--remove_file`

No the support for `--remove_file` is in #47 and is separate

**Review by mjonss — APPROVED** [review]
LGTM

File: src/main.go [hunk 1]
```diff
@@ -364,6 +364,8 @@ func (t *tester) Run() error {
 				q.Query = q.Query[:len(q.Query)-1]
 			}
 			t.disconnect(q.Query)
+		default:
+			log.WithFields(log.Fields{"command": q.firstWord, "arguments": q.Query, "line": q.Line}).Warn("unknown command")
```
Comment (mjonss) [inline src/main.go:368]: I did not get any output when running with `--log-level=warning` with a command like `--skip_if_hypergraph`, maybe better to also have something like this:
```
diff --git a/src/query.go b/src/query.go
index 5f6a4f5..6a0cf37 100644
--- a/src/query.go
+++ b/src/query.go
@@ -17,6 +17,7 @@ import (
        "strings"
 
        "github.com/pingcap/errors"
+       log "github.com/sirupsen/logrus"
 )
 
 var ErrInvalidCommand = errors.New("Found line beginning with -- that didn't contain a valid mysqltest command, check your syntax or use # if you intended to write comment")
@@ -193,6 +194,7 @@ func (q *query) getQueryType(qu string) error {
                        q.Query = qu
                        q.tp = Q_QUERY
                } else {
+                       log.Errorf("Line %d: --%s %s\n", q.Line, q.firstWord, q.Query)
                        return ErrInvalidCommand
                }
        }
```

**Review by mjonss — APPROVED** [review]
A suggestion for unknown Q_COMMENT_WITH_COMMAND

File: src/main.go (hunk 1 above)
Comment (dveeden) [reply src/main.go]: There are actually two cases:

- **Unimplemented commands**: Commands defined in `commandMap`, but not implemented in `Run()`.
- **Unknown commands**: Commands not defined in `commandMap`

I've updated the code to incorporate your patch and differentiate between the two cases

```
[dvaneeden@dve-carbon mysql-tester]$ cat t/unimplemented_command.test
--echo Testing for unknown commands
--skip_if_hypergraph foo bar baz
[dvaneeden@dve-carbon mysql-tester]$ cat t/unknown_command.test 
--echo Testing for unimplemented commands
--let $a=5
[dvaneeden@dve-carbon mysql-tester]$ cat r/unknown_command.result 
Testing for unimplemented commands
[dvaneeden@dve-carbon mysql-tester]$ ./mysql-tester unknown_command
INFO[0000] running tests: [unknown_command]             
WARN[0000] Create new db&{0 0xc000010098 0 {0 0} [0xc000288000] map[] 0 1 0xc000032180 false map[0xc000288000:map[0xc000288000:true]] map[] 0 0 0 0 <nil> 0 0 0 0 0x49d0a0} 
WARN[0000] command not implemented                       arguments=" $a=5" command=let line=2
./t/unknown_command.test: ok! 0 test cases passed, take time 2.3349e-05 s
INFO[0000] run test [unknown_command] ok                

Great, All tests passed
[dvaneeden@dve-carbon mysql-tester]$ ./mysql-tester unimplemented_command
INFO[0000] running tests: [unimplemented_command]       
WARN[0000] Create new db&{0 0xc000010010 0 {0 0} [0xc0000ae000] map[] 0 1 0xc000032060 false map[0xc0000ae000:map[0xc0000ae000:true]] map[] 0 0 0 0 <nil> 0 0 0 0 0x49d0a0} 
ERRO[0000] invalid command                               arguments=" foo bar baz" command=skip_if_hypergraph line=2
FATA[0000] run test [unimplemented_command] err: Found line beginning with -- that didn't contain a valid mysqltest command, check your syntax or use # if you intended to write comment 
```

**Comment by morgo** [comment]
> Are you sure? This only adds logging, it shouldn't change what's in the output file. I doesn't halt the test execution.

I see. It's only a log event, so it should be fine.

**Review by morgo — APPROVED** [review]
LGTM

**Review by mjonss — APPROVED** [review]
LGTM

**Comment by mjonss** [comment]
@dveeden please resolve the merge conflict.

**Comment by dveeden** [comment]
> @dveeden please resolve the merge conflict.

done

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**